salt/orch/kubernetes.sls

```diff
@@ -135,6 +135,40 @@ cert-setup:
     - require:
       - update-mine-again
 
+# restart salt minions if cert changes
+{%- set nodes_down = salt.saltutil.runner('manage.down') %}
+{%- if nodes_down|length >= 1 %}
+  {%- set is_responsive_node_tgt = 'not L@' + nodes_down|join(',') %}
+{%- else %}
+  {%- set is_responsive_node_tgt = '*' %}
+{%- endif %}
+{%- set is_salt_tgt = is_responsive_node_tgt + ' and not ca' %}
+{%- set salts = salt.caasp_nodes.get_with_expr(is_salt_tgt) %}
+
+salt-minion-restart:
+  salt.function:
+    - tgt: '{{ is_salt_tgt }}'
+    - tgt_type: compound
+    - name: service.restart
+    - arg:
+      - 'salt-minion'
+    - onchanges:
+      - cert-setup
+
+salt-minion-wait-for-start:
+  salt.wait_for_event:
+    # TODO: should this specify node_id instead of '*'?
+    - name: salt/minion/*/start
```

Comment:
It might be a good idea to iterate over known minion ids, I'm not sure if this will work as expected after a node is removed from the cluster (CI does not include this step)

Reply:
The ID List just below waits for the list of responsive nodes calculated just above, so it only waits for up nodes. :)